### ISSUE TITLE
SecurityGroups never finish creating

### ISSUE THREAD

**Issue by kferrone** [crossplane/provider-aws#296]
### What happened?
SecurityGroups never finish. The status Ready is always false. I have waited for quite some time and it does not seem to change. 

### How can we reproduce it?
Try this out:
```yaml
apiVersion: ec2.aws.crossplane.io/v1beta1
kind: SecurityGroup
metadata:
  name: foo-tester
spec:
  forProvider:
    description: A test sg to see if it can finish
    egress:
    - fromPort: -1
      ipProtocol: "-1"
      ipRanges:
      - cidrIp: 0.0.0.0/0
        description: Any IP address.
      toPort: -1
    groupName: foo-tester
    ingress:
    - fromPort: 6379
      ipProtocol: tcp
      ipRanges:
      - cidrIp: 10.101.0.0/16
        description: Just the cidr from the vpc only.
      toPort: 6379
    vpcIdRef:
      name: my-vpc
  providerRef:
    name: aws-provider
  reclaimPolicy: Delete
```

### What environment did it happen in?
Crossplane version: 0.12.0
Provider-aws: 0.11.0

**Comment by hasheddan** [comment]
@kferrone thanks for opening this! Any logs or status you could share with us? Do you see the security group in the AWS console?

**Comment by kferrone** [comment]
I do see them in the AWS console. From AWS side they are created perfectly fine. There are no useful logs in Crossplane controller. I searched and it all looks like it is working just not finishing.

**Comment by hasheddan** [comment]
Is the external name annotation present?

**Comment by kferrone** [comment]
It is indeed present

**Comment by muvaf** [comment]
@kferrone Thanks for reporting! There seems to be two issues regarding the up-to-date that is caused by using `-1`. It's fixed with #298
